Framework/DataHandling/src/LoadMuonNexusV2.cpp

```diff
@@ -0,0 +1,227 @@
+  declareProperty(std::make_unique<FileProperty>(
+                      "Filename", "", FileProperty::Load, extensions),
+                  "The name of the Nexus file to load");
+
+  declareProperty(
+      std::make_unique<WorkspaceProperty<Workspace>>("OutputWorkspace", "",
+                                                     Direction::Output),
+      "The name of the workspace to be created as the output of the\n"
+      "algorithm. For multiperiod files, one workspace will be\n"
+      "generated for each period");
+
+  auto mustBePositive = std::make_shared<BoundedValidator<int64_t>>();
+  mustBePositive->setLower(0);
+  declareProperty("SpectrumMin", static_cast<int64_t>(0), mustBePositive);
+  declareProperty("SpectrumMax", static_cast<int64_t>(EMPTY_INT()),
+                  mustBePositive);
+  declareProperty(std::make_unique<ArrayProperty<int64_t>>("SpectrumList"));
+  declareProperty("EntryNumber", static_cast<int64_t>(0), mustBePositive,
+                  "0 indicates that every entry is loaded, into a separate "
+                  "workspace within a group. "
+                  "A positive number identifies one entry to be loaded, into "
+                  "one workspace");
+
+  std::vector<std::string> FieldOptions{"Transverse", "Longitudinal"};
+  declareProperty("MainFieldDirection", "Transverse",
+                  std::make_shared<StringListValidator>(FieldOptions),
+                  "Output the main field direction if specified in Nexus file "
+                  "(default longitudinal).",
+                  Direction::Output);
+
+  declareProperty("TimeZero", 0.0,
+                  "Time zero in units of micro-seconds (default to 0.0)",
+                  Direction::Output);
+  declareProperty("FirstGoodData", 0.0,
+                  "First good data in units of micro-seconds (default to 0.0)",
+                  Direction::Output);
+
+  declareProperty(
+      std::make_unique<WorkspaceProperty<Workspace>>(
+          "DeadTimeTable", "", Direction::Output, PropertyMode::Optional),
+      "Table or a group of tables containing detector dead times.");
+
+  declareProperty(std::make_unique<WorkspaceProperty<Workspace>>(
+                      "DetectorGroupingTable", "", Direction::Output,
+                      PropertyMode::Optional),
+                  "Table or a group of tables with information about the "
+                  "detector grouping.");
+}
+void LoadMuonNexusV2::execLoader() {
+  this->setRethrows(true);
+  // prepare nexus entry
+  m_entrynumber = getProperty("EntryNumber");
+  m_filename = getPropertyValue("Filename");
+  NXRoot root(m_filename);
+  NXEntry entry = root.openEntry("raw_data_1");
+
+  // Execute child algorithm LoadISISNexus2 and load Muon specific properties
+  runLoadISISNexus();
+  loadMuonProperties(entry);
+
+  // Check if single or multi period file and create appropriate loading
+  // strategy
+  isEntryMultiPeriod(entry);
+  Workspace_sptr outWS = getProperty("OutputWorkspace");
```

Comment:
This looks a little unusual at first but I think it is grabbing the output workspace from `LoadISISNexus`?. Might it be clearer to have something like
```c++
auto outWS = runLoadISISNexus();
```
so that `runLoadISISNexus` just returns its pointer and it's a little easier to see where that workspace comes from?